### PR TITLE
docs: add v16 update guide

### PR DESCRIPTION
**Pull request by jelbourn**

### CODE REVIEW

File: .pullapprove.yml [hunk 1]
```diff
@@ -951,6 +951,7 @@ groups:
           'aio/content/guide/versions.md',
           'aio/content/guide/update-to-version-14.md',
           'aio/content/guide/update-to-version-15.md'
```
Comment (alxhub) [inline .pullapprove.yml:953]:
```suggestion
          'aio/content/guide/update-to-version-15.md',
```

File: aio/content/guide/update-to-version-16.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+You can try this new build setup by updating your `angular.json`:
+
+```json
+"architect": {
+  "build": {
+    "builder": "@angular-devkit/build-angular:browser-esbuild",
+```
+
+### Standalone component migration tooling
```
Comment (alxhub) [inline aio/content/guide/update-to-version-16.md:63]: We could also mention `ng new --standalone` here.

File: aio/content/guide/update-to-version-16.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+#### Angular Signals <span class="api-status-label dev-preview">developer preview</span>
+
+This release includes the first developer preview of Angular's new reactivity primitives: `signal`,
+`computed`, and `effect`. See the [signals guide](/guide/signals) for details and the
+[Angular Signals RFC](https://github.com/angular/angular/discussions/49685) for more background.
+
+#### Enhanced hydration <span class="api-status-label dev-preview">developer preview</span>
+
+Previously, when Angular bootstrapped on a page that was server-side rendered or compile-time
```
Comment (alxhub) [inline aio/content/guide/update-to-version-16.md:42]: Is it worth mentioning that this is opt-in?

**Review by alxhub — APPROVED** [review]

File: aio/content/guide/update-to-version-16.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+<!-- markdownLint-disable MD001 -->
+
+#### Angular Signals developer preview
+
+This release includes the first developer preview of Angular's new reactivity primitives: `signal`,
+`computed`, and `effect`. See the [signals guide](/guide/signals) for details and the
+[Angular Signals RFC](https://github.com/angular/angular/discussions/49685) for more background.
+
+#### Enhanced hydration developer preview
```
Comment (jessicajaniuk) [inline aio/content/guide/update-to-version-16.md:40]: IMHO enhanced is not the right word to use here. We should probably just say "Non-Destructive hydration developer preview".

**Review by jessicajaniuk — APPROVED** [review]
One suggestion, but otherwise LGTM

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 109eb88c6a3e7e321aab7320697d9c963363787b.

**Pull request closed without merging**